### PR TITLE
change streaming server url

### PR DESCRIPTION
**Pull request by tymmesyde**
#168

### CODE REVIEW

**Review by nklhtv — CHANGES_REQUESTED** [review]
Thank you for the PR!
Although its working and the code is great i don't think it provides good UX. It's not clear at all that user must hit enter key after changeing the url.
What i am proposing is to change the url input to a button that opens modal dialog with input field and submit button. 
Check "Add addon" button in addons route. It has the exact same behavior i've described.

**Comment by tymmesyde** [comment]
Yeah, it wasn't ideal..
I think I managed to add your proposal, but I'm not sure about my implementation.
Maybe a dedicated modal dialog like `AddonDetailsModal` would be nice ? Since `addAddonModal` and `editServerUrlModal` _(the one I added)_ are similar.

**Comment by pedroslopez** [comment]
Hi! What's the progress on this? I'd really love this feature

EDIT: After testing this, it seems to work fine. Only thing I would change is the text on the button to save the new streaming server url. At the moment it says "Edit", but I would change it to something like "Save".

File: src/routes/Settings/Settings.js [hunk 1]
```diff
@@ -21,6 +21,42 @@ const Settings = () => {
...
+                action: 'Ctx',
+                args: {
+                    action: 'UpdateSettings',
+                    args: {
+                        ...profile.settings,
+                        streaming_server_url: editServerUrlInputRef.current.value
+                    }
+                }
+            });
+            if (typeof closeEditServerUrlModal === 'function') {
```
Comment (nklhtv) [inline src/routes/Settings/Settings.js]: no need to check if that is a function here

File: src/routes/Settings/Settings.js [hunk 2]
```diff
@@ -21,6 +21,42 @@ const Settings = () => {
...
+        return [
+            {
+                className: styles['cancel-button'],
+                label: 'Cancel',
+                props: {
+                    onClick: closeEditServerUrlModal
+                }
+            },
+            {
+                label: 'Edit',
```
Comment (nklhtv) [inline src/routes/Settings/Settings.js]: That sould be named `Submit`

File: src/routes/Settings/styles.less [hunk 3]
```diff
@@ -275,9 +280,72 @@
                                 }
                             }
                         }
+
+                        &.edit-container {
+                            padding: 0;
+
+                            .label {
+                                overflow: hidden;
```
Comment (nklhtv) [inline src/routes/Settings/styles.less]: everything is `overflow: hidden` by default

File: src/routes/Settings/Settings.js [hunk 4]
```diff
@@ -354,10 +391,11 @@ const Settings = () => {
                             <div className={styles['option-name-container']}>
                                 <div className={styles['label']}>Url</div>
                             </div>
-                            <div className={classnames(styles['option-input-container'], styles['info-container'], styles['selectable'])}>
-                                <div className={styles['label']}>
-                                    {profile.settings.streaming_server_url}
-                                </div>
+                            <div className={classnames(styles['option-input-container'], styles['edit-container'])}>
+                                <div className={styles['label']}>{profile.settings.streaming_server_url}</div>
```
Comment (nklhtv) [inline src/routes/Settings/Settings.js]: add title attribute

File: src/routes/Settings/styles.less [hunk 5]
```diff
@@ -11,6 +11,11 @@
     multiselect-label: label;
 }
 
+:import('~stremio/common/ModalDialog/styles.less') {
+    addon-modal-content: modal-dialog-content;
```
Comment (nklhtv) [inline src/routes/Settings/styles.less]: that should be named streaming-server-modal-content

File: src/routes/Settings/Settings.js [hunk 6]
```diff
@@ -354,10 +391,11 @@ const Settings = () => {
...
+                                <Button className={classnames(styles['option-input-container'], styles['button-container'])} title={'Edit'} onClick={openEditServerUrlModal}>
```
Comment (nklhtv) [inline src/routes/Settings/Settings.js]: this is not a good place for reuse settings grid class names
use `configure-button-container` class name and put all of its styles there: flex, display, align-items, justify-content, background-color, width&height: 3.5rem etc

File: src/routes/Settings/styles.less [hunk 7]
```diff
@@ -275,9 +280,72 @@
...
+
+                            .label {
+                                overflow: hidden;
+                                white-space: pre;
+                                text-overflow: ellipsis;
+                                padding: 0 1rem;
+                            }
+
+                            .button-container {
+                                flex: auto;
```
Comment (nklhtv) [inline src/routes/Settings/styles.less]: use `flex: none;` so it could remain square

**Review by nklhtv — CHANGES_REQUESTED** [review]
I liked the changes in useProfileSettingsInputs.Could you bring them back? You could name the generic input like `serverUrlInput` with `value` and `onChange(value)` props, so in `editServerUrlOnSubmit` you will call `serverUrlInput.onChange(editServerUrlInputRef.current.value)` and use `serverUrlInput.value` in the JSX code

**Review by swetlasg — APPROVED** [review]

**Review by nklhtv — APPROVED** [review]

**Comment by nklhtv** [comment]
Fixes #168

**Pull request merged**